### ISSUE TITLE
src\vm\interoputil.cpp Line: 2862, GetNativeVariantForObject_Float_Success(obj: 3.14) [FAIL]

### ISSUE THREAD

**Issue by benaadams** [dotnet/coreclr#20373]
Running corefx against checked coreclr fails on Windows_NT x64 Checked Build and Test (Jit - CoreFx TieredCompilation=0) (seen in https://github.com/dotnet/coreclr/pull/19436#issuecomment-428962920)
```
Starting:    System.Runtime.InteropServices.Tests (parallel test collections = on, max threads = 4)
System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariantForObject_Float_Success(obj: 3.14) [FAIL]
 Assert.Equal() Failure
 Expected: 1078523331
 Actual:   -2459565879138388541
 Stack Trace:
   src\System.Runtime.InteropServices\tests\System\Runtime\InteropServices\Marshal\GetNativeVariantForObjectTests.cs(292,0):
   at System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariantForObject_Float_Success(Single obj)
 Assert failure(PID 14352 [0x00003810], Thread: 6240 [0x1860]): CONTRACT VIOLATION by GetValueTypeForGUID at 
  "d:\j\workspace\x64_checked_w---d7295605\src\vm\interoputil.cpp" @ 2862
MODE_COOPERATIVE encountered while thread is in preemptive state.

CONTRACT in GetValueTypeForGUID at "d:\j\workspace\x64_checked_w---d7295605\src\vm\interoputil.cpp" @ 2862
CONTRACT in OleVariant::MarshalRecordVariantOleToCom at "d:\j\workspace\x64_checked_w---d7295605\src\vm\olevariant.cpp" @ 2530
CONTRACT in OleVariant::MarshalComVariantForOleVariant at "d:\j\workspace\x64_checked_w---d7295605\src\vm\olevariant.cpp" @ 3447
CONTRACT in OleVariant::MarshalObjectForOleVariant at "d:\j\workspace\x64_checked_w---d7295605\src\vm\olevariant.cpp" @ 3042
CONTRACT in MarshalNative::GetObjectForNativeVariant at "d:\j\workspace\x64_checked_w---d7295605\src\vm\marshalnative.cpp" @ 1662
OVERRIDE_TYPE_LOAD_LEVEL_LIMIT in MethodDescCallSite::CallTargetWorker at "d:\j\workspace\x64_checked_w---d7295605\src\vm\callhelpers.cpp" @ 377
CONTRACT in MethodDescCallSite::CallTargetWorker at "d:\j\workspace\x64_checked_w---d7295605\src\vm\callhelpers.cpp" @ 367
CONTRACT in ThreadNative::KickOffThread_Worker at "d:\j\workspace\x64_checked_w---d7295605\src\vm\comsynchronizable.cpp" @ 194
CONTRACT in ManagedThreadBase_DispatchInner at "d:\j\workspace\x64_checked_w---d7295605\src\vm\threads.cpp" @ 8499
CONTRACT in ManagedThreadBase_FullTransitionWithAD at "d:\j\workspace\x64_checked_w---d7295605\src\vm\threads.cpp" @ 8847
CONTRACT in ThreadNative::KickOffThread at "d:\j\workspace\x64_checked_w---d7295605\src\vm\comsynchronizable.cpp" @ 293

We can't find the violated contract. Look for an old-style non-holder-based contract.


CORECLR! CONTRACT_ASSERT + 0x426 (0x00007ffd`623945e6)
CORECLR! EEContract::DoChecks + 0x277 (0x00007ffd`624a15df)
CORECLR! GetValueTypeForGUID + 0x28B (0x00007ffd`624bda97)
CORECLR! OleVariant::MarshalRecordVariantOleToCom + 0x405 (0x00007ffd`629ac3a5)
CORECLR! OleVariant::MarshalComVariantForOleVariant + 0x810 (0x00007ffd`6299f278)
CORECLR! OleVariant::MarshalObjectForOleVariant + 0xB2F (0x00007ffd`629a919f)
CORECLR! MarshalNative::GetObjectForNativeVariant + 0x3CF (0x00007ffd`62b7370f)
<no module>! <no symbol> + 0x0 (0x00007ffd`06272ae9)
<no module>! <no symbol> + 0x0 (0x00000053`9acbfa58)
<no module>! <no symbol> + 0x0 (0x00000053`9acbfa58)
    File: \src\vm\interoputil.cpp Line: 2862
    Image: \bin\testhost\netcoreapp-Windows_NT-Release-x64\dotnet.exe
```

**Comment by shimingsg** [comment]
similar failure in ci job : https://ci.dot.net/job/dotnet_coreclr/job/master/view/x64_arm64_altjit/job/jitstress/job/x64_arm64_altjit_checked_windows_nt_corefx_jitstressregs1/330


failed test:

System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariantForObject_Float_Success(obj: 3.14) [FAIL]

**Comment by BruceForstall** [comment]
This continues to happen for both Windows x86 and x64 (at least):

https://ci.dot.net/job/dotnet_coreclr/job/master/view/x64/job/jitstress/job/x64_checked_windows_nt_corefx_baseline/699/consoleFull
https://ci.dot.net/job/dotnet_coreclr/job/master/view/x86/job/jitstress/job/x86_checked_windows_nt_corefx_baseline/671/consoleFull

@jkotas @kouvel @RussKeldorph who can investigate this?

**Comment by jkotas** [comment]
I have submitted fix for the contract violation (#21238). The unexpected floating point result does not repro for me.

**Comment by AaronRobinsonMSFT** [comment]
I have verified the contract issue is fixed, but like @jkotas, the unexpected floating point result does not repo for me. Closing as resolved with #21238.
